File: src/main/java/com/hyungu/whoseturn/entities/UserEntity.java
package com.hyungu.whoseturn.entities;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "USERS")
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "MEMBER_ID", nullable = false)
    private String id;

    @Column(name = "NAME", nullable = false)
    private String name;

    @OneToMany(mappedBy = "USER_ID")
    private List<RoomUserEntity> roomUseris;

    public UserEntity() {
    }

    public UserEntity(String name) {
        this.name = name;
    }

    // getter
    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    // setter
    public void setName(String name) {
        this.name = name;
    }

}

File: src/main/kotlin/com/hyungu/whoseturn/entities/UserEntity.kt
package com.hyungu.whoseturn.entities

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import lombok.Data

@Entity(name = "users")
@Data
data class UserEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: String?,

    @ManyToMany
    @JoinTable(name = "room_user")
    var rooms: List<RoomEntity>,
)
